### ISSUE TITLE
OS X Unsatisfied Link Error w/ SNAP

### ISSUE THREAD

**Issue by mankoff** [bcdev/jpy#108]
When I run the snap-examples [snap-engine-python-operator](https://github.com/senbox-org/snap-examples/tree/master/snap-engine-python-operator) on OS X, I get the following error:

<img width="720" alt="screen shot 2017-09-20 at 09 19 17" src="https://user-images.githubusercontent.com/145117/30631584-077c3328-9de5-11e7-8c22-d829731fb5c0.png">

**Comment by forman** [comment]
@mankoff could you please send me your snappy configuration files (*.ini, *.properties).
Please also provide Python version, Java version, SNAP version.

**Comment by mankoff** [comment]
```
$ java -version
java version "1.8.0_144"
Java(TM) SE Runtime Environment (build 1.8.0_144-b01)
Java HotSpot(TM) 64-Bit Server VM (build 25.144-b01, mixed mode)

$ python --version
Python 3.4.5 :: Continuum Analytics, Inc.
```

SNAP version: `6.0-PREVIEW4`

Attached file generated with `zip snap-files.zip $(find ~/.snap -name "*.ini") $(find ~/.snap -name "*.properties")`

[snap-files.zip](https://github.com/bcdev/jpy/files/1320227/snap-files.zip)

**Comment by mankoff** [comment]
Although this bug was experienced by several other OS X users, who I *think* may have had SNAP 5.0, definitely different OS versions, and therefore probably different Java versions.

**Comment by forman** [comment]
Thanks! Afaik, we haven't seen this problem with various Pythons on various Linuxes (nor on Windows) so it may have something to do with the Anaconda distribution on OS X. The missing symbol `_PyBool_Type` should have already been a problem while compiling the jpy C-sources on Darwin. Strange...

**Comment by forman** [comment]
If your time allows for, would you mind building jpy on your own?

**Comment by mankoff** [comment]
I did build `jpy` using the following commands:

```
cd local/src/
git clone https://github.com/bcdev/jpy.git
cd jpy/
python setup.py bdist_wheel
cp dist/*.whl "~/.snap/snap-python/snappy"
```

**Comment by forman** [comment]
So this issue is about what happened *after* you built and used jpy?

**Comment by mankoff** [comment]
Yes. Sorry if that was not clear. This is a `snap` error. I built `jpy`. I built the example `snap` operator. When I try to run the operator in `snap` I get the error message show in this bug report.

**Comment by forman** [comment]
Weird! Sorry to bother you again, but could you please rebuild and try to find any warning or reference to `_PyBool_Type` in the compiler outputs.

**Comment by forman** [comment]
Selection of similar issues on OS X:

- https://github.com/trilinos/Trilinos/issues/569
- https://github.com/fontforge/fontforge/issues/2085
- https://svn.boost.org/trac10/ticket/7536

**Comment by forman** [comment]
Look what I found: #79
Was solved by setting `PYTHONHOME` to Python in use. Would you mind trying?

**Comment by mankoff** [comment]
Setting PYTHONHOME doesn't fix it for me, and introduces all sorts of new problems. I think, in general with Anaconda, this variable is not supposed to be set? Not sure.

The build script does report errors. See https://gist.github.com/mankoff/87fd62b7fc28ef806e798e7b8bcc4313

**Comment by forman** [comment]
The important message and root cause is `RuntimeError: can't find any Python shared library` and the debug outputs above that message show all the locations jpy is searching for that shared library. I'll try to find out, where it usually resides in an OS X Anaconda distribution...

**Comment by forman** [comment]
I currently don't have OS X access, I'll have a look tomorrow when I'm back in the office.

* https://conda.io/docs/user-guide/tasks/build-packages/use-shared-libraries.html
* https://groups.google.com/a/continuum.io/forum/#!topic/anaconda/fUpqjkK5jEI

**Comment by mankoff** [comment]
I think I improved the jpy build. See 2nd post (comment) here: https://gist.github.com/mankoff/87fd62b7fc28ef806e798e7b8bcc4313#gistcomment-2209441

I now `export DYLD_LIBRARY_PATH` to a path that contains the python.dylib.

However, after recompiling jpy and then `mvn clean package` for the test package, I still see the same Bool error in SNAP.

**Comment by forman** [comment]
Does `/Users/kdm/local/anaconda/envs/SNAP/lib/libpython3.4m.dylib` exist?

**Comment by mankoff** [comment]
Yes it does, which is why I think jpy seems to compile (better) now, with message, `DEBUG: No Python shared library file found in all search paths. Using fallback '/Users/kdm/local/anaconda/envs/SNAP/lib/libpython3.4m.dylib'`

**Comment by forman** [comment]
I just asked a colleague of mine if he can reproduce it ...

**Comment by vitbrando** [comment]
hi, 
I'm experiencing the same snap error.

$ uname -a
Darwin Kernel Version 17.6.0: Tue May  8 15:22:16 PDT 2018; root:xnu-4570.61.1~1/RELEASE_X86_64 x86_64

$ java -version
java version "1.8.0_161"
Java(TM) SE Runtime Environment (build 1.8.0_161-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.161-b12, mixed mode)

$ python --version
Python 3.4.5 :: Anaconda 2.3.0 (x86_64)

$ env |grep PY
PYTHONPATH=˜.snap/snap-python:˜~/coral-bu/c/work_in_progress/coding/py/swampy/sambuca_core:˜~/coral-bu/c/work_in_progress/coding/py/swampy/sambuca:˜anaconda/lib/pithon3.4/site-packages:˜anaconda/lib/pithon3.4:˜anaconda/lib:
PYTHOHOME=˜anaconda/lib/python3.4/

what should I try?
